File: web/src/krate/llmchat/dto.ts
import type {
  LLMChatBot,
  LLMChatRecord,
  LLMChatSession,
  LLMChatTemplate,
} from "./po";
import type { TID } from "@/lib/id_util";

export type LLMChatBotListRsp = {
  bots: LLMChatBot[];
};

export type LLMChatTemplateListRsp = {
  templates: LLMChatTemplate[];
};

export type LLMChatSessionListRsp = {
  sessions: LLMChatSession[];
};

export type LLMChatSessionRecordFetchRsp = {
  session?: LLMChatSession;
  records: LLMChatRecord[];
};

export type LLMChatSessionTruncateReq = {
  session_otid: TID;
  remove_rid_included: TID;
};

export type LLMChatBotCommitReq = {
  bot: LLMChatBot;
};

export type LLMChatBotCommitRsp = object;

export type LLMChatTemplateCommitReq = {
  template: LLMChatTemplate;
};

export type LLMChatTemplateCommitRsp = object;

export type LLMChatSessionCommitReq = {
  session: LLMChatSession;
};

export type LLMChatSessionCommitRsp = object;

export type LLMChatRecordCommitReq = {
  record: LLMChatRecord;
};

export type LLMChatRecordCommitRsp = object;

export type LLMChatBotListReq = object;

export type LLMChatTemplateListReq = object;

export type LLMChatSessionListReq = {
  session_otid?: TID;
};

export type LLMChatSessionRecordTruncateReq = {
  remove_rid_included: TID;
  session_otid: TID;
};

export type LLMChatSessionRecordTruncateRsp = {
  count: number;
};

export type LLMChatSessionRecordFetchReq = {
  session_otid: TID;
};

export type LLMChatBotArchiveReq = {
  bot_otid: TID;
};

export type LLMChatBotArchiveRsp = object;

export type LLMChatTemplateArchiveReq = {
  template_otid: TID;
};

export type LLMChatTemplateArchiveRsp = object;

export type LLMChatSessionArchiveReq = {
  session_otid: TID;
};

export type LLMChatSessionArchiveRsp = object;

export type LLMChatUpdateSessionReq = {
  title?: string;
  delete?: boolean;
  session_otid: TID;
};
export type LLMChatUpdateSessionRsp = object;

File: lib/backend/src/krate/llmchat/dto.rs
use chin_sql::time_type::TID;
use serde::{Deserialize, Serialize};

use super::*;

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct LLMChatBotCommitReq {
    pub bot: LLMChatBot,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct LLMChatBotCommitRsp {}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct LLMChatTemplateCommitReq {
    pub template: LLMChatTemplate,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct LLMChatTemplateCommitRsp {}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct LLMChatSessionCommitReq {
    pub session: LLMChatSession,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct LLMChatSessionCommitRsp {}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct LLMChatRecordCommitReq {
    pub record: LLMChatRecord,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct LLMChatRecordCommitRsp {}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct LLMChatBotListReq {}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct LLMChatBotListRsp {
    pub bots: Vec<LLMChatBot>,
}
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct LLMChatTemplateListReq {}
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct LLMChatTemplateListRsp {
    pub templates: Vec<LLMChatTemplate>,
}
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct LLMChatSessionListReq {
    pub session_otid: Option<TID>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct LLMChatSessionListRsp {
    pub sessions: Vec<LLMChatSession>,
}
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct LLMChatUpdateSessionReq {
    pub title: Option<String>,
    pub delete: Option<bool>,
    pub session_otid: TID,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct LLMChatUpdateSessionRsp {}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct LLMChatSessionRecordTruncateReq {
    pub remove_otid_included: TID,
    pub session_otid: TID,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct LLMChatSessionRecordTruncateRsp {
    pub count: usize,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct LLMChatSessionRecordFetchReq {
    pub session_otid: TID,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct LLMChatSessionRecordFetchRsp {
    pub session: Option<LLMChatSession>,
    pub records: Vec<LLMChatRecord>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct LLMChatBotArchiveReq {
    pub bot_otid: TID,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct LLMChatBotArchiveRsp {}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct LLMChatTemplateArchiveReq {
    pub template_otid: TID,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct LLMChatTemplateArchiveRsp {}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct LLMChatSessionArchiveReq {
    pub session_otid: TID,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct LLMChatSessionArchiveRsp {}
